### PR TITLE
SONAR-6256 Add dependencies in batch report

### PR DESCRIPTION
**Pull request by julienlancelot**

### CODE REVIEW

File: sonar-batch-protocol/src/main/java/org/sonar/batch/protocol/output/FileStructure.java [hunk 1]
```diff
@@ -38,6 +38,8 @@
     COVERAGES("coverages-", Domain.PB),
     TESTS("tests-", Domain.PB),
     COVERAGE_DETAILS("coverage-details-", Domain.PB),
+    FILE_DEPENDENCIES("file-dependencies-", Domain.PB),
+    MODULE_DEPENDENCIES("module-dependencies-", Domain.PB),
```
Comment (teryk) [inline sonar-batch-protocol/src/main/java/org/sonar/batch/protocol/output/FileStructure.java]: where is the writer/reader for module dependencies ?

File: sonar-batch-protocol/src/main/protobuf/batch_report.proto [hunk 2]
```diff
@@ -251,3 +251,19 @@ message CoverageDetail {
     repeated int32 covered_line = 2 [packed = true];
   }
 }
+
+message FileDependency {
+  optional int32 to_file_ref = 1;
+  optional int32 weight = 2;
+}
+
+message ModuleDependencies {
```
Comment (teryk) [inline sonar-batch-protocol/src/main/protobuf/batch_report.proto]: no need for streaming ?

File: sonar-batch-protocol/src/main/java/org/sonar/batch/protocol/output/FileStructure.java (hunk 1 above)
Comment (julienlancelot) [inline sonar-batch-protocol/src/main/java/org/sonar/batch/protocol/output/FileStructure.java]: Oups, I've forgot it...

File: sonar-batch-protocol/src/main/protobuf/batch_report.proto (hunk 2 above)
Comment (julienlancelot) [inline sonar-batch-protocol/src/main/protobuf/batch_report.proto]: It's not possible to do some streaming as we need to read all data in memory to be able to compute the module dependencies.

**Comment by teryk** [comment]
apply my feedback and ...
lgtm-pulp-fiction

**Pull request merged**